src/core-asyncpipeline/AsyncPipeline/AsyncPipeline.T/ToValueTask/ToValueTask.cs

```diff
@@ -0,0 +1,13 @@
+#nullable enable
+
+using System.Threading.Tasks;
+
+namespace System
+{
+    partial struct AsyncPipeline<T>
+    {
+        public ValueTask<T> ToValueTask()
+            =>
+            isCanceled is false ? valueTask : ValueTask.FromCanceled<T>(CanceledToken());
```

Comment:
I suppose the internal task should not be open to a user and.
It looks like an abstraction leak allows the user to break the pipeline.
I seems the true conversion should be implemented, something like that:

```cs
isCanceled is false ? InnerToValueTask(valueTask) : ValueTask.FromCanceled<T>(CanceledToken());

private static async ValueTask<T> InnerToValueTask(ValueTask<T> valueTask)
    =>
    await valueTask.ConfigureAwait(false);

```

Reply:
Decided to keep as is.